### ISSUE TITLE
pylint infinite loop with a recursive method + comprehension list

### ISSUE THREAD

**Issue by moylop260** [PyCQA/pylint#952]
### Steps to reproduce

1. Save the following code in a file:

  ```python
def recursive_method(mylist_objs):
    ids2 = []
    if random.randint(1, 3) == 3:
        ids2 += recursive_method([x.value for x in mylist_objs])
    return ids2
```
2. Execute `pylint myfile.py`

### Current behaviour

- `RuntimeError: maximum recursion depth exceeded`

### Expected behaviour
 - pylint checks results.

### pylint --version output
```bash
No config file found, using default configuration
pylint 2.0.0,
astroid 1.5.0
Python 2.7.6 (default, Jun 22 2015, 17:58:13)
[GCC 4.8.2]
```

The shas are the last of master for astroid and pylint.

NOTE: Yeah the code is ugly this is just a example to report this issue.

**Comment by moylop260** [comment]
@PCManticore 
Do you make sense?

**Comment by PCManticore** [comment]
Yeah, we have some problems with inferring recursion. There are some opened tickets on astroid as well, regarding similar examples, but I didn't have time to come up with a good solution. I might try to address these after having a potential 2.0 release ready.

**Comment by moylop260** [comment]
Thank for reply
Then I'll close this issue and I'll follow the astroid ones
